azure/lock.sh

```diff
@@ -0,0 +1,14 @@
+# Lock related functions
+
+# azure_lock_create creates a new lock.
+# Available lock types are: "CanNotDelete", "NotSpecified", "ReadOnly".
+fn azure_lock_create(lockname, locktype, resgroup) {
+	az lock create --lock-type $locktype --name $lockname --resource-group $resgroup
+}
+
+# azure_lock_delete deletes a lock with the given name.
+fn azure_lock_delete(lockname, resgroup) {
+	az lock delete --name $lockname --resource-group $resgroup
+	# FIXME: delete is not sync, sometimes things may feel
```

Comment:
@tiago4orion I agree, just want to validate this whole idea before investing more on it